service/src/test/java/org/cbioportal/genome_nexus/util/GenomicVariantTest.java

```diff
@@ -0,0 +1,57 @@
+package org.cbioportal.genome_nexus.util;
+
+import static org.junit.Assert.assertEquals;
+import org.junit.Test;
+
+public class GenomicVariantTest {
+    @Test    
+    public void testHgvsSubstitutionToGenomicVariant() {
+        GenomicVariant variant = GenomicVariant.fromHgvs("17:g.41242962_41242963insGA");
+        GenomicVariantUtil util = variant.getUtil();
+
+        assertEquals(util.getChromosome(), "17");
+        assertEquals(util.getStart(), (Integer) 41242962);
+        assertEquals(util.getEnd(), (Integer) 41242963);
+        assertEquals(util.getRef(), "ins");
+        assertEquals(util.getAlt(), "GA");
+    }
+
+    @Test
+    public void testHgvsAdditionToGenomicVariant() {
+        GenomicVariant variant = GenomicVariant.fromHgvs("10:g.32867861_32867862insT");
+        GenomicVariantUtil util = variant.getUtil();
+
+        assertEquals(util.getChromosome(), "10");
+        assertEquals(util.getStart(), (Integer) 32867861);
+        assertEquals(util.getEnd(), (Integer) 32867862);
+        assertEquals(util.getRef(), "ins");
+        assertEquals(util.getAlt(), "T");
+
+    }
+
+    @Test
+
+    public void testHgvsDeletionToGenomicVariant() {
+        GenomicVariant variant = GenomicVariant.fromHgvs("1:g.19_21del");
```

Comment:
this example is actually quite tricky, because you don't know what reference allele is being deleted. Usually you'll see examples like:

```
1:g.19_21delACG
```
and then the reference allele becomes:
```
ACG
```
and alt becomes
```
-
```
or empty string if you prefer

Reply:
maybe we can do for the cases where we don't know what the three reference bases deleted were:
```
XXX
```